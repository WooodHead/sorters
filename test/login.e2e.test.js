import cheerio from 'cheerio'
import pretty from 'pretty'
import {MongoClient, ObjectID} from 'mongodb'
import {setup, teardown} from './setup'

let app
let db
let browser

beforeAll(async () => {
    const res = await setup()
    app = res.app
    db = res.db
    browser = res.browser
})

afterEach(async () => {
    await db.dropDatabase()    
})

afterAll(async () => {
    await teardown({db, app, browser})
})

afterAll(async () => {
    await db.close()

    await new Promise((resolve, reject) => {
        app.on('close', () => {
            resolve()
        })
        try {
            process.kill(-app.pid)
        } catch (e) {
            console.log(e)
        }
    })
})

describe('login', () => {
    it('displays', async () => {
        const browserPage = await browser.createPage()
        const status = await browserPage.open(`http://localhost:3000/login`)
        expect(status).toBe('success')

        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const html = pretty(page.html())
        expect(html).toMatchSnapshot()
    })
})
