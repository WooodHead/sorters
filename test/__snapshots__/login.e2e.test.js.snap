// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login displays 1`] = `
"<div data-reactroot=\\"\\">
  <div>
    <div style=\\"padding-top:50px;padding-bottom:20px\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\">
        <div class=\\"container\\">
          <div class=\\"navbar-header\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button>
            <a class=\\"navbar-brand\\" href=\\"/\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\">
            <ul class=\\"nav navbar-nav navbar-right\\">
              <li>
                <a href=\\"/\\">Home</a>
              </li>
              <li>
                <a href=\\"/about\\">About</a>
              </li>
              <li>
                <a href=\\"/news\\">News</a>
              </li>
              <li>
                <a href=\\"/users\\">Sorters</a>
              </li>
              <li class=\\"active\\">
                <a href=\\"/login\\">Log in</a>
              </li>
              <li>
                <a href=\\"/register\\">Register</a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div style=\\"max-width:300px;margin:auto\\">
      <h1>Log in</h1>
      <form>
        <div class=\\"form-group\\">
          <label for=\\"username\\">Username or Email</label>
          <input type=\\"string\\" class=\\"form-control\\" id=\\"username\\" placeholder=\\"Username\\">
        </div>
        <div class=\\"form-group\\">
          <label for=\\"password\\">Password</label>
          <input type=\\"password\\" class=\\"form-control\\" id=\\"password\\" placeholder=\\"******\\">
          <a href=\\"/forgot-password\\">Forgot your password?</a>
        </div>
        <div class=\\"form-group\\">
          <button type=\\"submit\\" class=\\"btn btn-primary btn-block\\">Log in</button>
        </div>
      </form>
      <p style=\\"text-align:center\\">or</p>
      <p>New user?
        <a href=\\"/register\\">register</a>.</p>
    </div>
    <footer>
      <div class=\\"container\\">
        <hr>
        <p>Brought to you with &#x2764; by
          <a href=\\"/about\\">Nick Redmark</a>. Support this platform on
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\">Patreon</a>.</p>
      </div>
    </footer>
  </div>
</div>"
`;
